src/components/building/building.js

```diff
@@ -40,77 +41,66 @@ export const Building = () => {
-    const buildInProgressTime =
-      currentUserBuildingInProgress.length === 0
-        ? ''
-        : timeBuilder(Math.round(currentUserBuildingInProgress[0].diffTime / 1000));
-
-    let buildPriceTextColor = [];
-    let notEnoughRescourceButtonDisable = false;
-
-    gameBuildingType.buildMaterials.forEach(buildMaterial => {
-      for (const key in currentUserResources) {
-        if (key === buildMaterial.resourceType) {
-          
-          const currentResource = currentUserResources[key];
-          const neededResource = buildingPrice(nextLevel, gameBuildingType.id, key);
-          const enoughRescource = currentResource >= neededResource;
-          
-          const textColor = enoughRescource ? 'black' : 'red';
-          if (!enoughRescource) {
-            notEnoughRescourceButtonDisable = true;
-          }
-          buildPriceTextColor.push(textColor);
-        }
-      }
-    });
-
-    return (
-      <article key={gameBuildingType.id}>
-        <Buildingcard
-          buildType={0}
-          buildId={gameBuildingType.id}
-          buildName={gameBuildingType.name}
-          buildDescription={gameBuildingType.description}
-          buildTime={buildTime}
-          buildPrice={gameBuildingType.buildMaterials}
-          buildPriceTextColor={buildPriceTextColor}
-          buildYield={gameBuildingType.productionMaterials}
-          buildInProgressTime={buildInProgressTime}
-          currentBuildLevel={currentUserBuilding.level}
-          buttonText={buttonTexts}
-          buttonDisabled={notEnoughRescourceButtonDisable || buildInProgressButtonDisable }
-          buttonFunction={onHandleClickUpgrade}
-        />
-      </article>
-    );
+  if (currentUserBuildingInProgress.length > 0) {
+    const inProgressId = currentUserBuildingInProgress[0].buildingId;
+    buttonTexts = inProgressId === gameBuildingType.id ? 'in Progress' : buttonTexts;
+  }
+
+  buttonTexts = currentUserBuilding.level > 0 && buttonTexts !== 'in Progress' ? 'Upgrade' : buttonTexts;
+
+  const nextLevel = currentUserBuilding.level + 1;
+
+  let buildPrice = [];
+  gameBuildingType.buildMaterials.forEach(material => {
+    if (material.resourceType === 'money' || material.resourceType === 'iron') {
+      buildPrice.push(buildingPrice(nextLevel, gameBuildingType.id, material.resourceType));
+    }
   });
+  const buildTime = buildingTime(
+    buildPrice[0] !== undefined ? buildPrice[0] : 0,
+    buildPrice[1] !== undefined ? buildPrice[1] : 0
+  );
+
+  const buildInProgressButtonDisable = currentUserBuildingInProgress.length === 0 ? false : true;
+  const buildInProgressTime =
+    currentUserBuildingInProgress.length === 0
+      ? ''
+      : timeBuilder(Math.round(currentUserBuildingInProgress[0].diffTime / 1000));
+
+      const checkEnoughRescource = checkRescource(gameBuildingType,currentUserResources,nextLevel)
+
+  return (
+    <article key={gameBuildingType.id}>
```

Comment:
Since this is not in a `map` anymore - I think 🤔 - you don't need the key here.
```suggestion
    <article>
```

Reply:
thanks done, i dont see it.